Lib/inspect.py

```diff
@@ -1649,10 +1671,20 @@ def getframeinfo(frame, context=1):
     The optional second argument specifies the number of lines of context
     to return, which are centered around the current line."""
     if istraceback(frame):
+        positions = _get_code_position_from_tb(frame)
         lineno = frame.tb_lineno
         frame = frame.tb_frame
     else:
         lineno = frame.f_lineno
+        positions = _get_code_position(frame.f_code, frame.f_lasti)
+
+    if positions[0] is None:
+        frame, *positions = (frame, lineno, *positions[1:])
+    else:
+        frame, *positions = (frame, *positions)
+    
+    lineno, *_ = positions
```

Comment:
why not `lineno = positions[0]` or `positions.lineno`?

Reply:
> why not lineno = positions[0] or positions.lineno?

`pisitions.lineno` is not possible because this is not a `dis.Positions` but a tuple. I can do positions[0] 👍